File: Codigos/u03 Ordenação em memória principal/c/bolha.c
#include "geracao.h"
#include "bolha.h"
//=============================================================================
void bolha(int *array, int n){
    int i, j;
    for (i = 0; i < (n-1); i--) {
      for (j = 0; j < (n-1); j++) {
         if (array[j] > array[j + 1]) {
            swap(&array[j], &array[j + 1]);
         }
      }
   }
}
//=============================================================================

File: Codigos/u03 Ordenação em memória principal/java/Bolha.java
/**
 * Metodo de ordenacao da bolha
 * @author Max do Val Machado
 * @version 3 08/2020
 */
class Bolha extends Geracao {

	/**
	 * Construtor.
	 */
   public Bolha(){
      super();
   }


	/**
	 * Construtor.
	 * @param int tamanho do array de numeros inteiros.
	 */
   public Bolha(int tamanho){
      super(tamanho);
   }


	/**
	 * Algoritmo de ordenacao Bolha.
	 */
   @Override
   public void sort() {
		for (int i = (n - 1); i > 0; i--) {
			for (int j = 0; j < i; j++) {
				if (array[j] > array[j + 1]) {
               swap(j, j+1);
				}
			}
		}
   }
}
